### DISCUSSION TITLE
Python 3.10 support

### DISCUSSION THREAD

**Discussion by Ilyazyk** [equinor/neqsim#463] [Release]
Dear Prof. Solbraa,
Thank you very much for your work!
I am doing very first steps and trying to get running your software (2.3.26) on my Ubuntu 22.04 with Python 3.10
Is Python version 3.10 supported?

The sample code [Pumps](https://colab.research.google.com/github/EvenSol/NeqSim-Colab/blob/master/notebooks/process/pumps.ipynb) works in Colab, but doesn't in my environment. 

The row of code 
**fluid1.setTotalFlowRate(135.0, "m3/hr")**
rises an error
"[main] ERROR neqsim.util.unit.RateUnit - unit not supported m3/hr". 

At the same time units 'kg/hr' work.
Some other examples work fine.
The only difference I've found so far is Python 3.7 is used in Colab

I am using OpenJDK 64-Bit Server VM (build 11.0.15+10-Ubuntu-0ubuntu0.22.04.1, mixed mode, sharing)
and python packages list is here [pip_list.txt](https://github.com/equinor/neqsim/files/9071891/pip_list.txt)

And one more question, please. 
Could you point me out where I can get a documentation like a whole list of supported components for a method "fluid1.addComponent() "?

**Comment by EvenSol** [comment]
The component names in the embedded database you can find in the link on the bottom on the page:
https://github.com/equinor/neqsim/wiki/The-NeqSim-parameter-database

**Reply by EvenSol** [reply]
Direct link to the component names:
https://github.com/equinor/neqsim/blob/master/src/main/resources/neqsim_component_names.txt

**Comment by EvenSol** [comment]
Use link to see the supported rate units (and conversion to reference mol/sec). m3/hr should be Am3/hr. However it is safer to use kg/hr (as you don’t need reference density). I will update the pump example. I guess the reason you don’t see the error in Colab, is that the error log is not routed to the output (so it is correct on your computer).

https://github.com/equinor/neqsim/blob/4632e45859511b236d2f0632aee0ae7112094e35/src/main/java/neqsim/util/unit/RateUnit.java#L64

**Comment by Ilyazyk** [comment]
Thank you for update, the links were useful.
[Am3/h] works fine. Thank you!
I'd prefer to use [Am3/h] and provide fluid1.setTotalFlowRate() with actual values of pressure and temperature, and let system do the job. Because it is what we usually get from sensors. In my experience it happens very rare that mass flowmeters are used. 
In existing implementation I could not get a value of density (for mass flow calculation based on volumetric flow) before the call of fluid1.setTotalFlowRate(). It doesn't seem correct as flow doesn't affect density.
And as a wish, I'd like to have an API documentation with all function parameters described (in addition to good existing examples).

**Reply by EvenSol** [reply]
I agree. The API documentation should have been improved.
I have updated the JavaDoc - so it reflects the main branch:

https://htmlpreview.github.io/?https://github.com/equinor/neqsimhome/blob/master/javadoc/site/apidocs/index.html
